src/Magnum/ImGuiIntegration/Integration.h

```diff
@@ -0,0 +1,197 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2018 ShaddyAQN <ShaddyAQN@gmail.com>
+    Copyright © 2018 lecopivo <skrivantomas@seznam.cz>
+    Copyright © 2018 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <Magnum/GL/AbstractShaderProgram.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/GL/Texture.h>
+#include <Magnum/GL/Buffer.h>
+#include <Magnum/GL/Mesh.h>
+#include <Magnum/Timeline.h>
+#include <Magnum/GL/Attribute.h>
+
+#include "Magnum/ImGuiIntegration/visibility.h"
+
+namespace Magnum { namespace ImGuiIntegration {
+
+namespace Implementation {
+
+class MAGNUM_IMGUIINTEGRATION_EXPORT ImGuiShader : public GL::AbstractShaderProgram {
+    public:
+        typedef GL::Attribute<0, Vector2> Position;
+        typedef GL::Attribute<1, Vector2> TextureCoordinates;
+        typedef GL::Attribute<2, Vector4> Color;
+
+        explicit ImGuiShader();
+
+        ImGuiShader& setProjectionMatrix(const Matrix4& matrix) {
+            setUniform(_projMatrixUniform, matrix);
+            return *this;
+        }
+
+        ImGuiShader& bindTexture(GL::Texture2D& texture) {
+            texture.bind(TextureLayer);
+            return *this;
+        }
+
+    private:
+        enum: Int {
+            TextureLayer = 0
+        };
+
+        Int _projMatrixUniform;
+};
+
+}
+
+/*
+@brief ImGui context
+
+Handles initialization and destruction of ImGui context.
+
+@section ImGuiIntegration-Context-usage Usage
+
+There should always only be one instance of @ref Context. As soon as this
+one instance is created, you can access it via @ref Context::get().
+
+Example:
+
+@code{.cpp}
+Context context;
+
+// ...
+
+if(Context::get().newFrame({640, 480}, {640, 480})) {
+    // ...
+}
+
+@endcode
```

Comment:
Would be great to have this snippet compiled -- put in a cpp file into `doc/snippets/`, mentioned in `doc/snippets/CMakeLists.txt` and referenced from here via `@snippet ImGuiIntegration.cpp something`. Look into BulletIntegration or OVR for reference.